### ISSUE TITLE
[LOW] [Performance] [$250] Investigate why unfocused screens are updated by onyx

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#42505]
Coming from onyx bump https://github.com/Expensify/App/pull/42057#issuecomment-2125620911

Investigate how to fix an issue where unfocused screen is being updated by onyx. There was a solution put in place to unblock the onyx update, however, we should explore if there are better ways to solve it. Slack discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1716406572009159)

To fix https://github.com/Expensify/App/pull/42057#issuecomment-2111103644, this commit was used https://github.com/Expensify/App/pull/42057/commits/e7e73c8e1c5fea51fc7d5d626afe066b007325c1

cc @chrispader 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f294fb9fd27db9de</li>
        <li>Upwork Job ID: 1793569593909895168</li>
        <li>Last Price Increase: 2024-05-23</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @DylanDylann</details>

**Comment by chrispader** [comment]
commenting for assignment 👋🏼

**Comment by mountiny** [comment]
thanks Chris

**Comment by mountiny** [comment]
@chrispader can you look into this once the deploy blocker is resolved?

**Comment by chrispader** [comment]
> @chrispader can you look into this once the deploy blocker is resolved?

yes, gonna do!

**Comment by isabelastisser** [comment]
Any updates? @DylanDylann @chrispader Thanks!

**Comment by mountiny** [comment]
I believe Chris is now focusing on fixing latest onyx version which is higher priority so no updates still

**Comment by isabelastisser** [comment]
@mountiny should I put this on hold?

**Comment by chrispader** [comment]
i might start working on this tmrw, depends how much there is to do in https://github.com/Expensify/App/pull/42772

**Comment by isabelastisser** [comment]
@mountiny, which wave and release should I add this to? thanks!

**Comment by mountiny** [comment]
Added as low to the newdot quality as this is performance related

**Comment by isabelastisser** [comment]
Thanks, Vit!

**Comment by isabelastisser** [comment]
@chrispader @DylanDylann, any progress?

**Comment by chrispader** [comment]
Going to work on this now and over the weekend!

**Comment by chrispader** [comment]
I now finished profiling and investigating some of (navigation) flows in E/App and it seems this issue was fixed by the recent [Onyx bump PR](https://github.com/Expensify/App/pull/42772).

I think the reason for old screen props being updated before, was that Onyx was in a weird intermediate state of some transitions, that are now finished by version `2.0.48`.

The following issues are still fixed without the need for any changes:

- https://github.com/Expensify/App/pull/42057#issuecomment-2111103644
- https://github.com/Expensify/App/pull/42713

Therefore i think we can close this out!
